pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/TransactionAggregator.java

```diff
@@ -36,6 +38,11 @@
 @Slf4j
 public class TransactionAggregator {
 
+    /**
+     * Used for tracking duplicate TYPE definitions.
+     */
+    static Map<String, String> metricWithTypeDefinition = new HashMap<>();
```

Comment:
This should be set as a thread-local variable, otherwise multiple instances of brokers will clash (eg: during unit tests)